astarte/device/device_mqtt.py

```diff
@@ -517,8 +559,36 @@ def __setup_subscriptions(self) -> None:
             f"{self.__get_base_topic()}/control/consumer/properties", qos=2
         )
         for interface in self._introspection.get_all_server_owned_interfaces():
+            self.__subscribe_interface(interface)
+
+    def __subscribe_interface(self, interface: Interface) -> None:
+        """
+        Subscribe an interface to a topic from the MQTT broker.
+
+        This interface only performs the subscription for server owned interfaces.
+
+        Parameters
+        ----------
+        interface: Interface
+            The server owned interface to use for the subscription.
+        """
+        if interface.is_server_owned():
```

Comment:
Just nitpicking but in my opinion the if could be moved out to make imediately clear that we're only subscribing to server owned interfaces. And at that point we could remove the function since i'm seeing it is only used in one place.

Reply:
Agreed with this and the following comments.